plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/Realtime.tsx

```diff
@@ -0,0 +1,150 @@
+import { Icon, Switch } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { BaseFieldProps } from "../types";
+
+const Realtime: React.FC<BaseFieldProps<"realtime">> = ({ value, editMode, onUpdate }) => {
+  const [intervalInSecond, setIntervalValue] = useState<number>(value.updateInterval ?? 30);
+  const [timer, setTimer] = useState<string>("00:00:00");
+  const [enableUpdate, changeUpdateState] = useState<boolean>(true);
+  const Ref = useRef<any>(null);
+
+  const getTimeRemaining = useCallback((interval: number) => {
+    const seconds = Math.floor(interval % 60);
+    const minutes = Math.floor((interval / 60) % 60);
+    const hours = Math.floor((interval / 60 / 60) % 24);
+    if (interval > 0) {
+      setTimer(
+        (hours > 9 ? hours : "0" + hours) +
+          ":" +
+          (minutes > 9 ? minutes : "0" + minutes) +
+          ":" +
+          (seconds > 9 ? seconds : "0" + seconds),
+      );
+    }
+  }, []);
+
+  const startTimer = useCallback(
+    (interval: number) => {
+      if (Ref.current) clearInterval(Ref.current);
+      const id = setInterval(() => {
+        if (enableUpdate) {
+          getTimeRemaining(interval);
+          interval--;
+          if (interval === 0) interval = intervalInSecond;
+        }
+      }, 1000);
+
+      Ref.current = id;
+    },
```

Comment:
To be honest I can't tell if this'll work or not. Is this tested? If its tested and working, the only thing that would be nice is the const id be renamed to perhaps remainingTime, or just added directly like Ref.current = setInterval.....

Reply:
yes it is tested , and done for the renaming

Reply:
c9e5706fc4bb3535ff518c320d648fb4f2ea4b1b